ext/dcos-installer/dcos_installer/cli_dispatcher.py

```diff
@@ -0,0 +1,168 @@
+        pp = PrettyPrint(result)
+        pp.stage_name = action.__name__
+        pp.beautify('print_data')
+
+    finally:
+        loop.close()
+    exitcode = 0
+    for host_result in result:
+        for command_result in host_result:
+            for host, process_result in command_result.items():
+                if process_result['returncode'] != 0:
+                    exitcode += 1
+    print_header('ACTION {} COMPLETE'.format(action.__name__))
+    pp.print_summary()
+    return exitcode
+
+
+def tall_enough_to_ride():
+    choices_true = ['yes', 'y']
+    choices_false = ['no', 'n']
+    while True:
+        do_uninstall = input('This will uninstall DC/OS on your cluster. You may need to manually remove '
+                             '/var/lib/zookeeper in some cases after this completes, please see our documentation '
+                             'for details. Are you ABSOLUTELY sure you want to proceed? [ (y)es/(n)o ]: ')
+        if do_uninstall.lower() in choices_true:
+            return
+        elif do_uninstall.lower() in choices_false:
+            sys.exit(1)
+        else:
+            log.error('Choices are [y]es or [n]o. "{}" is not a choice'.format(do_uninstall))
+
+
+def log_warn_only():
+    """Drop to warning level and down to get around gen.generate() log.info
+    output"""
+    coloredlogs.install(
+        level='WARNING',
+        level_styles={
+            'warn': {
+                'color': 'yellow'
+            },
+            'error': {
+                'color': 'red',
+                'bold': True,
+            },
+        },
+        fmt='%(message)s',
+        isatty=True
+    )
+
+
+def print_validation_errors(messages):
+    log.error("Validation of configuration parameters failed: ")
+    for key, message in messages.items():
+        log.error('{}: {}'.format(key, message))
+
+
+def validate_ssh_config_or_exit():
+    validation_errors = backend.do_validate_ssh_config()
+    if validation_errors:
+        print_validation_errors(validation_errors)
+        sys.exit(1)
+
+
+def dispatch_option(args):
+    """Dispatches ClI options which are not installer actions"""
+    def web(args):
+        print_header("Starting DC/OS installer in web mode")
+        async_server.start(args)
+
+    def hash_password(args):
+        if len(args.hash_password) > 0:
+            print_header("HASHING PASSWORD TO SHA512")
+            backend.hash_password(args.hash_password)
+            sys.exit(0)
+
+    def validate_config(args):
+        print_header('VALIDATING CONFIGURATION')
+        log_warn_only()
+        validation_errors = backend.do_validate_gen_config()
+        if validation_errors:
+            print_validation_errors(validation_errors)
+            sys.exit(1)
+        sys.exit(0)
+
+    def genconf(args):
+        print_header("EXECUTING CONFIGURATION GENERATION")
+        code = backend.do_configure()
+        if code != 0:
+            sys.exit(1)
+        sys.exit(0)
```

Comment:
Can drop this `sys.exit(0)` to the bottom of the parent def(), and only call once.